src/__tests__/union.test.ts

```diff
@@ -91,6 +153,59 @@ describe('union', () => {
                     done();
                 });
             });
+
+            it('callbacks are only called once', done => {
+                const vol = Volume.fromJSON({
+                    '/foo/bar': 'bar',
+                });
+                const vol2 = Volume.fromJSON({
+                    '/foo/bar': 'baz',
+                });
+                
+                const ufs = new Union() as any;
+                ufs.use(vol as any);
+                ufs.use(vol2 as any);
+
+                const mockCallback = jest.fn();
+                ufs.readFile("/foo/bar", "utf8", () => {
+                    mockCallback();
+                    expect(mockCallback).toBeCalledTimes(1);
```

Comment:
Shall we  remove these two lines? As they seem to effectively not do anything.

Reply:
These two are actually a test that I made while checking my implementations of the async iterator which ended up calling the callback multiple times. Since I’ve moved to my own implementation of readdir it does feel redundant now though.